ga4gh/datamodel/genotype_phenotype.py

```diff
@@ -0,0 +1,467 @@
+"""
+Module responsible for translating g2p data into GA4GH native
+objects.
+"""
+
+import ga4gh.protocol as protocol
+import ga4gh.exceptions as exceptions
+
+import rdflib
+from rdflib import URIRef
+from rdflib import Graph
+
+
+class G2PDataset:
+    """
+    an rdf object store
+    """
+
+    # keep the graph as a static variable
+    g = None
+    # overwrite tests/unit/cgd-test.ttl
+    record_test = False
+    #
+    g2 = Graph()
+
+    def __init__(self, is_prod):
+
+        self.search_query = """
+            PREFIX OBAN: <http://purl.org/oban/>
+            PREFIX OBO: <http://purl.obolibrary.org/obo/>
+            PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+            PREFIX faldo: <http://biohackathon.org/resource/faldo#>
+            PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+            SELECT %PROPERTIES%
+                WHERE {
+                    ?s    a OBAN:association .
+                    ?s  OBAN:association_has_subject ?l .
+                    ?l rdfs:label ?location_label  .
+                    %LOCATION%
+                    ?s  OBO:RO_has_environment  ?drug .
+                    ?drug  rdfs:label ?drug_label  .
+                    ?s  OBAN:association_has_object  ?d .
+                    ?d  rdfs:label ?disease_label  .
+                    ?d rdf:type ?disease .
+                    ?s  OBAN:association_has_object_property  ?evidence .
+                    OPTIONAL {  ?evidence  rdfs:label ?evidence_label } .
+              %FILTER%
+            }
+            """
+
+        if (G2PDataset.g):
+            return
+        G2PDataset.g = rdflib.Graph()
+
+        if (is_prod):
+            # larger data set for exploration
+            # "http://nif-crawler.neuinfo.org/monarch/ttl/cgd.ttl"
+            G2PDataset.g.parse('tests/unit/cgd.ttl', format="n3")
+
+            # supplementary
+            G2PDataset.g.parse('https://raw.githubusercontent.com/oborel/'
+                               'obo-relations/master/ro.owl',
+                               format="xml")
+
+        else:
+            # small data set for unit Tests
+            G2PDataset.g.parse('tests/unit/cgd-test.ttl', format="n3")
+
+    def _search(self, request):
+        # from IPython.core.debugger import Pdb
+        # Pdb().set_trace()
+        offset = 0
+        if request.pageToken:
+            print("pageToken " + request.pageToken)
+            offset = request.pageToken.split(':')[1]
+            print("offset " + offset)
+
+        associations = self.query_labels(request.feature,
+                                         request.evidence,
+                                         request.phenotype,
+                                         request.pageSize,
+                                         offset
+                                         )
+        # from IPython.core.debugger import Pdb ;        Pdb().set_trace()
+
+        self.associations_length = len(associations)
```

Comment:
All instance variables should be declared in the constructor, be camelCase and be protected by default, i.e.
``self._associationsLength``.